--- lttng-ust-2.7.1/liblttng-ust-python-agent/Makefile.am.orig	2016-01-25 22:11:19.823939945 +0100
+++ lttng-ust-2.7.1/liblttng-ust-python-agent/Makefile.am	2016-01-25 22:12:22.277270532 +0100
@@ -15,7 +15,7 @@
 	$(PYTHON) setup.py build --verbose
 
 install-exec-local:
-	@opts="--prefix=$(prefix) --verbose --no-compile $(DISTSETUPOPTS)"; \
+	@opts="--prefix=$(prefix) --install-purelib=$(pythondir) --verbose --optimize=2 $(DISTSETUPOPTS)"; \
 	if [ "$(DESTDIR)" != "" ]; then \
 		opts="$$opts --root=$(DESTDIR)"; \
 	fi; \
