--- lttng-ust-2.4.0/liblttng-ust-java/Makefile.am.orig	2014-03-03 08:55:13.364604698 +0100
+++ lttng-ust-2.4.0/liblttng-ust-java/Makefile.am	2014-03-03 08:57:12.334599705 +0100
@@ -9,7 +9,7 @@
 liblttng_ust_java_la_SOURCES = LTTngUst.c lttng_ust_java.h
 nodist_liblttng_ust_java_la_SOURCES = org_lttng_ust_LTTngUst.h
 dist_noinst_DATA = $(LTTNG_JUST_SRCDIR)/LTTngUst.java
-liblttng_ust_java_la_LIBADD = -lc -L$(top_builddir)/liblttng-ust/.libs -llttng-ust
+liblttng_ust_java_la_LIBADD = $(top_builddir)/liblttng-ust/liblttng-ust.la -ldl
 
 if HAVE_JAVA_JDK
 JCC=$(JAVA_JDK)/bin
--- lttng-ust-2.4.0/liblttng-ust/Makefile.am.orig	2014-02-28 19:20:10.000000000 +0100
+++ lttng-ust-2.4.0/liblttng-ust/Makefile.am	2014-03-03 10:55:00.270969760 +0100
@@ -67,6 +67,7 @@
 	$(top_builddir)/libringbuffer/libringbuffer.la
 
 liblttng_ust_la_LIBADD = \
+	-ldl \
 	-lpthread \
 	-lrt \
 	-lurcu-cds \
--- lttng-ust-2.4.0/liblttng-ust-jul/Makefile.am.orig	2014-02-28 19:20:10.000000000 +0100
+++ lttng-ust-2.4.0/liblttng-ust-jul/Makefile.am	2014-03-03 10:55:41.377634701 +0100
@@ -15,7 +15,7 @@
 	$(LTTNG_JUL_SRCDIR)/LTTngSessiondCmd2_4.java \
 	$(LTTNG_JUL_SRCDIR)/LTTngTCPSessiondClient.java \
 	$(LTTNG_JUL_SRCDIR)/LTTngThread.java
-liblttng_ust_jul_jni_la_LIBADD = -lc -L$(top_builddir)/liblttng-ust/.libs -llttng-ust
+liblttng_ust_jul_jni_la_LIBADD = $(top_builddir)/liblttng-ust/liblttng-ust.la -ldl
 
 lttnglibjava_DATA = liblttng-ust-jul.jar
 
