--- lttng-ust-2.13.1/src/lib/lttng-ust-java/Makefile.am.orig	2021-12-09 20:45:58.000000000 +0100
+++ lttng-ust-2.13.1/src/lib/lttng-ust-java/Makefile.am	2022-02-05 10:59:11.838865926 +0100
@@ -29,7 +29,7 @@ org_lttng_ust_LTTngUst.h: $(jarfile)
 else
 org_lttng_ust_LTTngUst.h: jni-header.stamp
 
-jni-header.stamp: $(dist_noinst_JAVA)
+jni-header.stamp: classnoinst.stamp
 	$(JAVAH) -classpath $(srcdir) $(JAVAHFLAGS) org.lttng.ust.LTTngUst && \
 	echo "JNI header generated" > jni-header.stamp
 endif
