--- lttng-ust-2.12.1/liblttng-ust-java/Makefile.am.orig	2021-02-26 18:55:34.153038775 +0100
+++ lttng-ust-2.12.1/liblttng-ust-java/Makefile.am	2021-02-26 18:58:12.021196975 +0100
@@ -26,7 +26,7 @@
 else
 org_lttng_ust_LTTngUst.h: jni-header.stamp
 
-jni-header.stamp: $(dist_noinst_JAVA)
+jni-header.stamp: classnoinst.stamp
 	$(JAVAH) -classpath $(srcdir) $(JAVAHFLAGS) org.lttng.ust.LTTngUst && \
 	echo "JNI header generated" > jni-header.stamp
 endif
