--- lttng-ust-2.8.1/python-lttngust/Makefile.am.orig	2016-05-31 06:38:27.000000000 +0200
+++ lttng-ust-2.8.1/python-lttngust/Makefile.am	2016-06-25 12:26:23.159661719 +0200
@@ -6,7 +6,7 @@
 	$(PYTHON) setup.py build --verbose
 
 install-exec-local:
-	@opts="--prefix=$(prefix) --verbose --no-compile $(DISTSETUPOPTS)"; \
+	@opts="--prefix=$(prefix) --install-purelib=$(pythondir) --verbose --optimize=2 $(DISTSETUPOPTS)"; \
 	if [ "$(DESTDIR)" != "" ]; then \
 		opts="$$opts --root=$(DESTDIR)"; \
 	fi; \
